pkg/reconciler/pipelinerun/controller.go

```diff
@@ -54,7 +59,9 @@ func NewController(opts *pipeline.Options, clock clock.PassiveClock, tracerProvi
 		pipelineRunInformer := pipelineruninformer.Get(ctx)
 		resolutionInformer := resolutioninformer.Get(ctx)
 		verificationpolicyInformer := verificationpolicyinformer.Get(ctx)
-		configStore := config.NewStore(logger.Named("config-store"), pipelinerunmetrics.MetricsOnStore(logger))
+		tracerProvider := tracing.New(TracerProviderName)
+		//nolint:contextcheck // OnStore methods does not support context as a parameter
```

Comment:
rather than disabling the linter, would it make sense for OnStore to accept a ctx?

Reply:
Nope. It only makes sense if the OnStore method is directly consuming the context object.

But in this case, the OnStore method returns a callback function that is called every time the configmap changes. Ideally the callback method should be accepting the context object which is passed by the caller. But this logic is implemented inside knative library and we cannot change it.

That's why i had to create a context object inside the callback method. The linter is reporting a false positive error in this case. Of course we can pass the context and use it inside the callback, but it doesn't much sense to do it that way.